src/Generators/Ics.php

```diff
@@ -55,4 +66,107 @@ protected function generateEventUid(Link $link): string
     {
         return md5($link->from->format(\DateTime::ATOM).$link->to->format(\DateTime::ATOM).$link->title.$link->address);
     }
+
+    protected function addVTimezoneComponents(VCalendar $vcalendar, array $timeZones, \DateTime $from, \DateTime $to)
+    {
+        foreach ($timeZones as $timeZone) {
+            /* @var \DateTimeZone $timeZone */
+            if ($timeZone->getName() !== 'UTC') {
+                $vcalendar->add(
+                    $this->generateVTimeZoneComponent(
+                        $vcalendar,
+                        $timeZone,
+                        $from->getTimestamp(),
+                        $to->getTimestamp()
+                    )
+                );
+            }
+        }
+    }
+
+    /**
+     * Returns a VTIMEZONE component for a Olson timezone identifier
+     * with daylight transitions covering the given date range.
+     *
+     * Kinldy inspired from https://gist.github.com/thomascube/47ff7d530244c669825736b10877a200
+     * and https://stackoverflow.com/a/25971680
+     *
+     * @param VCalendar $vcalendar
+     * @param \DateTimeZone $timeZone Timezone
+     * @param int $from Unix timestamp with first date/time in this timezone
+     * @param int $to Unix timestap with last date/time in this timezone
+     *
+     * @return Component A Sabre\VObject\Component object representing a VTIMEZONE definition
+     */
+    protected function generateVTimeZoneComponent(VCalendar $vcalendar, \DateTimeZone $timeZone, int $from = 0, int $to = 0)
+    {
+        if (! $from) {
+            $from = time();
+        }
+        if (! $to) {
+            $to = $from;
+        }
+
+        // get all transitions for one year back/ahead
+        $year = 86400 * 360;
+        $transitions = $timeZone->getTransitions($from - $year, $to + $year);
+
+        $vTimeZone = $vcalendar->createComponent('VTIMEZONE');
+        $vTimeZone->TZID = $timeZone->getName();
+
+        $std = null;
+        $dst = null;
+        foreach ($transitions as $i => $trans) {
+            $component = null;
+
+            if ($i === 0) {
+                // remember the offset for the next TZOFFSETFROM value
+                $tzfrom = $trans['offset'] / 3600;
+            }
+
+            // daylight saving time definition
+            if ($trans['isdst']) {
+                $t_dst = $trans['ts'];
+                $dst = $vcalendar->createComponent('DAYLIGHT');
+                $component = $dst;
+            } // standard time definition
+            else {
+                $t_std = $trans['ts'];
+                $std = $vcalendar->createComponent('STANDARD');
+                $component = $std;
+            }
+
+            if ($component) {
```

Comment:
this condition is extra: `$component` is always a `Component` instance (see if-else above)